### ISSUE TITLE
Autogenerate documenation for the client

### ISSUE THREAD

**Issue by Zelldon** [zeebe-io/zeebe-client-csharp#43]
**Is your feature request related to a problem? Please describe.**
It would be useful to read to doc in more humand readable form on a page.
Take a look at https://github.com/dotnet/docfx

Related https://forum.zeebe.io/t/c-zeebe-client-methods/758/6

**Comment by Zelldon** [comment]
This blogpost might help
https://www.codeproject.com/Articles/1260150/How-To-Integrate-DocFX-API-Documentation-Into-Your

Today I totally failed to generate the docs, but there were several issues like not detailed doc (funny right) and most things are written for windows.

But I think the following steps should work
```
Dowload zip
Unzip
Cd zeebe repo
Mono $pathDocfx.exe init
Set the src to /**.csproj
Cd docfx_project
Mono $pathDocfx.exe metadata # <-- this was the missing part
THEN mono $pathDocfx.exe build --serve
```
Generating metadata was the missing part which should create the docs from the source files. I will try this tomorrow

**Comment by Zelldon** [comment]
Better guide http://www.hardkoded.com/blog/creating-docfx-site

**Comment by Zelldon** [comment]
I had a long fight with the docfx, mono, msbuild and my linux machine. After several hours/days. I decided to go with a different tool. One which I already know. I will go with [doxygen](http://www.doxygen.nl/) for now it is quite easy to use and provides a clean documentation. This should be fine for now.

With the following commits:
 * https://github.com/zeebe-io/zeebe-client-csharp/commit/870182bca8749791eb284cf4ad235ec0506ddcc3
 * https://github.com/zeebe-io/zeebe-client-csharp/commit/29cd4ac2ce7a0256759c1b701a4801ac1905c193

I added the new documentation which is now available under https://zeebe-io.github.io/zeebe-client-csharp/index.html
